RecoLocalTracker/SiPhase2VectorHitBuilder/src/VectorHitBuilderAlgorithm.cc

```diff
@@ -0,0 +1,574 @@
+#include "RecoLocalTracker/SiPhase2VectorHitBuilder/interface/VectorHitBuilderAlgorithm.h"
+#include "Geometry/CommonDetUnit/interface/GeomDet.h"
+#include "Geometry/CommonTopologies/interface/PixelTopology.h"
+#include "DataFormats/TrackerRecHit2D/interface/VectorHit2D.h"
+
+bool VectorHitBuilderAlgorithm::LocalPositionSort::operator()(Phase2TrackerCluster1DRef clus1,
+                                                              Phase2TrackerCluster1DRef clus2) const {
+  static std::map<std::pair<Phase2TrackerCluster1DRef, Phase2TrackerCluster1DRef>, bool> cache;
+
+  if (cache.find(std::make_pair(clus1, clus2)) != cache.end())
+    return cache[std::pair(clus1, clus2)];
+  else if (cache.find(std::make_pair(clus2, clus1)) != cache.end())
+    return !cache[std::pair(clus1, clus2)];
+
+  const PixelGeomDetUnit* gdu1 = dynamic_cast<const PixelGeomDetUnit*>(geomDet_);
+  auto&& lparams1 = cpe_->localParameters(*clus1, *gdu1);  // x, y, z, e2_xx, e2_xy, e2_yy
+  auto&& lparams2 = cpe_->localParameters(*clus2, *gdu1);  // x, y, z, e2_xx, e2_xy, e2_yy
+
+  cache[std::make_pair(clus1, clus2)] = lparams1.first.x() < lparams2.first.x();
+  return lparams1.first.x() < lparams2.first.x();
+  ;
+}
+
+void VectorHitBuilderAlgorithm::run(edm::Handle<edmNew::DetSetVector<Phase2TrackerCluster1D>> clusters,
+                                    VectorHitCollectionNew& vhAcc,
+                                    VectorHitCollectionNew& vhRej,
+                                    edmNew::DetSetVector<Phase2TrackerCluster1D>& clustersAcc,
+                                    edmNew::DetSetVector<Phase2TrackerCluster1D>& clustersRej) const {
+  LogDebug("VectorHitBuilderAlgorithm") << "Run VectorHitBuilderAlgorithm ... \n";
+  const edmNew::DetSetVector<Phase2TrackerCluster1D>* ClustersPhase2Collection = clusters.product();
+
+  std::unordered_map<DetId, std::vector<VectorHit>> tempVHAcc, tempVHRej;
+  std::unordered_map<DetId, std::vector<VectorHit>>::iterator it_temporary;
+
+  //loop over the DetSetVector
+  LogDebug("VectorHitBuilderAlgorithm") << "with #clusters : " << ClustersPhase2Collection->size() << std::endl;
+  for (auto DSViter : *ClustersPhase2Collection) {
```

Comment:
no , no, no
ok loop over all but select only the upper, by definition the lower will be skipped

Reply:
Point taken, I added the continue below for the lower case.